src/pydicom/dataelem.py

```diff
@@ -208,13 +211,27 @@ def __init__(
             tag = Tag(tag)
         self.tag = tag
 
+        # check if the value is a buffer. If so, we can't check the length or access
+        # the value directly
+        is_buffered = False
+        if isinstance(value, BufferedIOBase):
+            is_buffered = True
+            valid_vrs = (VR_.OB, VR_.OB_OW, VR_.OW)
```

Comment:
Can you add this as a set in valuerep instead? We try and keep all the VR stuff together to make maintenance easier when new VRs are added

Reply:
Maybe consider `BYTE_VRS - VR.UN` instead?

Reply:
I think it'd be good to support OD and OF as well for *Float Pixel Data* and *Double Float Pixel Data*.

Possibly the other O* VRs?

Reply:
Great idea - will update.